File: rb/features.rb
# OecPro SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module OecProFeatures
  def self.make_feature(name)
    case name
    when "base"
      OecProBaseFeature.new
    when "ratelimit"
      OecProRatelimitFeature.new
    when "retry"
      OecProRetryFeature.new
    when "test"
      OecProTestFeature.new
    when "timeout"
      OecProTimeoutFeature.new
    else
      OecProBaseFeature.new
    end
  end
end

File: py/features.py
# OecPro SDK feature factory

from feature.base_feature import OecProBaseFeature
from feature.test_feature import OecProTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OecProBaseFeature(),
        "test": lambda: OecProTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
